### DISCUSSION TITLE
Customize console to OAuth redirect URL

### DISCUSSION THREAD

**Discussion by linearlyIndependent** [okd-project/okd#901] [Q&A] [answered]
**Describe the bug**

The OKD cluster is running behind a firewall/reverse-proxy at _<**internal**.address.lan>_. In the cluster Ingress, spec.domain is apps._<**internal**.address.lan>_ and spec.appsDomain is apps._<**external**.address.tld>_. The console and OAuth are both accessible over the internal and the external address over corresponding Routes.  Traffic gets forwarded from _<**external**.address.tld>_ to  _<**internal**.address.lan>_. Everything works fine except the redirection from the console to OAuth.

When accessing the console (without being logged in) at console-openshift-console.apps._<**external**.address.tld>_, the user is redirected to oauth-openshift.apps. _<**internal**.address.lan>_ instead of oauth-openshift.apps._<**external**.address.tld>_.

Is there any way to configure this redirection? I could not find anything in the docs or in the Red Hat Knowledgebase.

**Version**

4.7.0-0.okd-2021-07-03-190901

**How reproducible**

100%

**How to reproduce**

- OKD cluster deployed on KVM VMs (FCOS), behind a pfSense firewall and HAProxy. 
- Configure  the external hostname [according to the documentation](https://docs.openshift.com/container-platform/4.7/web_console/customizing-the-web-console.html#customizing-the-web-console-url_customizing-web-console)
  - This creates also the corresponding Route to expose the console.  
- Configure a Route to expose OAuth on the external hostname (same as the default oauth-openshift Route, just with the external hostname).
- In a fresh browser session (user is not logged in yet) open console-openshift-console.apps.<external.address.tld>, which will redirect you to oauth-openshift.apps.<internal.address.lan> for login.

**Log bundle**

**Comment by vrutkovs** [comment, marked as answer]
See https://github.com/openshift/okd/discussions/750 - this is not supported

**Reply by linearlyIndependent** [reply]
Thank you for the answer!
